### ISSUE TITLE
Which parameters should have traces?

### ISSUE THREAD

**Issue by Howuhh** [mohmdelsayed/streaming-drl#12]
Hi! Great work. Sorry if this is a naive question. I do not know the theory in detail, but should all parameters have traces? It seems like this would not make much sense for biases, layer or batch normalization parameters, embeddings, etc.

What would be the correct way to handle this?

Thanks!

**Comment by mohmdelsayed** [comment]
Hi @Howuhh,
This is a great question!

Eligibility traces are, by definition, traces of past value or log π gradients with respect to the learnable parameters. From that perspective, one could argue that any learnable weight should, in principle, have an associated trace.

That said, eligibility traces in combination with neural networks are still relatively understudied. It may be the case that maintaining traces for certain parameters is more meaningful or beneficial than for others, but at present, there are no clear theoretical or empirical results supporting this.

Overall, this remains an open question, and I think it would make for a very interesting research direction to explore.

Thanks again for raising this question.
